### PR TITLE
Checking rebuild trigger condition based in replica create time

### PR DESCRIPTION
**Pull request by mynktl**
Changes:
-  Checking rebuild trigger condition based in replica create time
 If there are no IOs in healthy or all other replica queues older
  than `replica->create_time` then rebuild for that replica will get
  triggered else it will be delayed.

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

### CODE REVIEW

File: src/replication.c [hunk 1]
```diff
@@ -491,6 +491,70 @@ start_rebuild(void *buf, replica_t *replica, uint64_t data_len)
...
+	struct timespec diff1, diff2 = { 0 }, now;
+	int ret = 0;
+	replica_t *r;
+	rcmd_t *cmd;
+
+	ASSERT(MTX_LOCKED(&spec->rq_mtx));
+
+	TAILQ_FOREACH(r, &spec->rq, r_next) {
+		if ((replica && r == replica) || (!replica)) {
+			cmd = TAILQ_FIRST(&r->readyq);
```
Comment (vishnuitta) [inline src/replication.c:522]: there are chances that this 'cmd' might have got deleted in replica_thread.

File: src/replication.c [hunk 2]
```diff
@@ -491,6 +491,70 @@ start_rebuild(void *buf, replica_t *replica, uint64_t data_len)
...
+			if (cmd != NULL) {
+				timesdiff(CLOCK_MONOTONIC, cmd->queued_time,
+				    now, diff1);
```
Comment (vishnuitta) [inline src/replication.c:525]: 'now' is not initialized

Comment (vishnuitta) [reply src/replication.c:525]: ok.. timesdiff takes care of it.. but, we need to use a fixed time to know the difference

File: src/replication.c (hunk 1 above)
Comment (mynktl) [reply src/replication.c:522]: yes.

File: src/replication.c (hunk 2 above)
Comment (mynktl) [reply src/replication.c:525]: ok

**Pull request closed without merging**